### PR TITLE
For padding capability, use the output type to compute the modified extents

### PR DESCRIPTION
**Pull request by yasahi-hpc**
This PR aims to use the output type to compute the modified extents.
For `C2R` transform, the modified shape is computed differently, 
and thus we need to know whether the transform is `C2R` or not.
Previously, a boolean `is_C2R` is given to `get_modified_shape` function, 
but `is_C2R` is checked internally from input and output view types in this function. 

Following modifications are made:
1.  `is_C2R` is checked internally in `get_modified_shape` function
2. Tests are parameterized over output types. (Previously, we used parameterized over boolean `is_C2R`)

### CODE REVIEW

**Review by cedricchevalier19 — COMMENTED** [review]
Some thoughts, I've to learn more about FFT to be pertinent!

**Comment by yasahi-hpc** [comment]
> Some thoughts, I've to learn more about FFT to be pertinent!

Thank you for a quick review. 
In case FFT knowledge is needed, I will try to explain the details in PR.

File: common/src/KokkosFFT_padding.hpp
Comment (yasahi-hpc) [reply common/src/KokkosFFT_padding.hpp:15]: Then pass by values?

Comment (yasahi-hpc) [reply common/src/KokkosFFT_padding.hpp:15]: Sure. I will add some comments of this function. 
This function is designed to give a new shape based on the user request. 
If the input shape is larger than the current view, then a new view with larger shape is allocated with zero padding. 

> Furthermore, are you sure that get_modified_shape is not too generic and will not be useful anywhere else in the project?

This function is for that purpose only. However, the data allocation function is generic which can be used for padding, transpose buffer, and so on.

**Comment by yasahi-hpc** [comment]
@cedricchevalier19
May I have your second review please

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:111]:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:137]:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:164]:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:190]:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:217]:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:244]:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:317]:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:354]:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:391]:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:428]:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:466]:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:504]:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.

File: common/unit_test/Test_Padding.cpp
Comment (cedricchevalier19) [inline common/unit_test/Test_Padding.cpp:564]:
```suggestion
  test_reshape2D_3DView<float_type>();
```

**Comment by yasahi-hpc** [comment]
> A typo that should be fixed to run the correct test.
> 
> Else, it is mostly cosmetic suggestions.

Thank you. I have fixed accordingly. If you approve the changes, I will merge this

**Review by cedricchevalier19 — APPROVED** [review]
Looks good.

**Pull request merged**
